internal/core/services/service.go

```
@@ -0,0 +1,143 @@
+package services
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/k8s-container-integrity-monitor/internal/core/models"
+	"github.com/k8s-container-integrity-monitor/internal/core/ports"
+	"github.com/k8s-container-integrity-monitor/internal/repositories"
+	"github.com/k8s-container-integrity-monitor/pkg/api"
+
+	"github.com/sirupsen/logrus"
+)
+
+type AppService struct {
+	ports.IHashService
+	ports.IAppRepository
+	ports.IKuberService
+	logger *logrus.Logger
+}
+
+// NewAppService creates a new struct AppService
+func NewAppService(r *repositories.AppRepository, algorithm string, logger *logrus.Logger) (*AppService, error) {
+	algorithm = strings.ToUpper(algorithm)
+	IHashService, err := NewHashService(r.IHashRepository, algorithm, logger)
+	if err != nil {
+		return nil, err
+	}
+	kuberService := NewKuberService(logger)
+	return &AppService{
+		IHashService:   IHashService,
+		IAppRepository: r,
+		IKuberService:  kuberService,
+		logger:         logger,
+	}, nil
+}
+
+//GetPID getting pid by process name
+func (as *AppService) GetPID(configData models.ConfigMapData) (int, error) {
+	if os.Chdir(os.Getenv("PROC_DIR")) != nil {
+		fmt.Println("/proc unavailable.")
```

maybe we need to return this error if method signature explains that we will return error if it will appear during execution?

---

returned an error